drivers/watchdog/wdt_sdw_apb.c

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright 2022 Broadcom
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT synopsys_sdw_watchdog
+
+/**
+ * @brief Driver for sdw APB Watchdog.
+ */
+
+#include <errno.h>
+#include <drivers/watchdog.h>
+
+struct wdog_sdw_apb {
+	/* offset: 0x000 (r/w) watchdog control register */
+	volatile uint32_t  ctrl;
+	/* offset: 0x004 (r/ ) watchdog timeout range register */
+	volatile uint32_t  torr;
+	/* offset: 0x008 (r/w) watchdog current conter value register */
+	volatile uint32_t  ccvr;
+	/* offset: 0x00c ( /w) watchdog counter restart register */
+	volatile uint32_t  crr;
+	/* offset: 0x010 (r/ ) watchdog interrupt status register */
+	volatile uint32_t  intstat;
+	/* offset: 0x014 (r/ ) watchdog interrupt clear register */
+	volatile uint32_t  eoi;
+	/* offset: 0x01c (r/ ) watchdog protection level register */
+	volatile uint32_t  prot_level;
+	volatile uint32_t  reserved0[49];
+	/* offset: 0xe4 (r/ ) watchdog component parameter 5 register */
+	volatile uint32_t  comp_param_5;
+	/* offset: 0xe8 (r/ ) watchdog component parameter 4 register */
+	volatile uint32_t  comp_param_4;
+	/* offset: 0xec (r/ ) watchdog component parameter 3 register */
+	volatile uint32_t  comp_param_3;
+	/* offset: 0xf0 (r/ ) watchdog component parameter 2 register */
+	volatile uint32_t  comp_param_2;
+	/* offset: 0xf4 (r/ ) watchdog component parameter 1 register */
+	volatile uint32_t  comp_param_1;
+	/* offset: 0xf8 (r/ ) watchdog component version register */
+	volatile uint32_t  comp_version;
+	/* offset: 0xfc (r/ ) watchdog component type register */
+	volatile uint32_t  comp_type;
+};
+
+#define WDOG_STRUCT \
+	((volatile struct wdog_sdw_apb *)(DT_INST_REG_ADDR(0)))
```

Comment:
Why is this a single-instance driver? Is it because it will only be supported on SoCs which have one instance of this IP block? If so, please add a build-assert that the number of enabled instances is at most 1. If not, please make this a multi-instance driver instead.